### PR TITLE
UI: Adds user informations after authentication

### PR DESCRIPTION
**Pull request by pratap0007**
This patch displays user informations such as
`github profile picture` and `user name`
after successfully authentication

Screenshot from 2021-03-31 22-21-05

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/api/index.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ export interface Api {
   setRating(resourceId: number, token: string, rating: number): Promise<void | null>;
   getRefreshToken(refreshToken: string): Promise<ITokenInfo>;
   getAccessToken(accessToken: string): Promise<ITokenInfo>;
+  info(token: string): Promise<IUserInfo>;
```
Comment (vinamra28) [inline ui/src/api/index.ts:38]: I guess we can rename this to userInfo. WDYT?

File: ui/src/store/auth.ts [hunk 2]
```diff
@@ -193,6 +206,23 @@ export const AuthStore = types
         };
         self.setErrorMessage(error);
       }
+    }),
+
+    getInfo: flow(function* () {
```
Comment (vinamra28) [inline ui/src/store/auth.ts:211]: same here. We can make it as getUserInfo

File: ui/src/store/auth.ts [hunk 3]
```diff
@@ -29,6 +36,7 @@ export const AuthStore = types
   .model('AuthStore', {
     accessTokenInfo: types.optional(TokenInfo, {}),
     refreshTokenInfo: types.optional(TokenInfo, {}),
+    info: types.optional(UserInfo, {}),
```
Comment (vinamra28) [inline ui/src/store/auth.ts:39]: same userInfo

File: ui/src/api/index.ts (hunk 1 above)
Comment (pratap0007) [reply ui/src/api/index.ts:38]: we have `user` as AuthStore instnace in the root store so here I have used `info`. if I rename it to `userInfo` then `user.userInfo` might be a bit confusing , WDYT?

Comment (PuneetPunamiya) [reply ui/src/api/index.ts:38]: Can we rename it to `profile` which will call the function as `user.Profile`

File: ui/src/store/auth.ts (hunk 3 above)
Comment (PuneetPunamiya) [reply ui/src/store/auth.ts:39]: Are we not changing here to profile ?

**Review by PuneetPunamiya — COMMENTED** [review]
I think if we are using `profile` then I guess we should add it everywhere in order to maintain consistency

File: ui/src/store/auth.ts (hunk 3 above)
Comment (pratap0007) [reply ui/src/store/auth.ts:39]: just missed :smile: will update

File: ui/src/store/auth.test.ts [hunk 4]
```diff
@@ -166,4 +166,31 @@ describe('Store functions', () => {
       }
     );
   });
+
+  it('can get user information after logged in', (done) => {
+    const store = AuthStore.create(
```
Comment (PuneetPunamiya) [inline ui/src/store/auth.test.ts:171]: Can we use the variable name `user` here instead of `store`

File: ui/src/store/auth.ts [hunk 5]
```diff
@@ -46,6 +54,11 @@ export const AuthStore = types
       self.refreshTokenInfo.expiresAt = item.expiresAt;
       self.refreshTokenInfo.refreshInterval = item.refreshInterval;
     },
+    addUserInfo(item: IUserInfo) {
```
Comment (PuneetPunamiya) [inline ui/src/store/auth.ts:57]:
```suggestion
    addUserProfile(item: IUserProfile) {
```

File: ui/src/store/auth.ts [hunk 6]
```diff
@@ -8,12 +8,19 @@ export const TokenInfo = types.model('TokenInfo', {
...
+  avatarUrl: types.optional(types.string, '')
+});
+
 export const Error = types.model('Error', {
   status: types.optional(types.number, 0),
   customMessage: types.optional(types.string, ''),
   serverMessage: types.optional(types.string, '')
 });
 
+export type IUserInfo = Instance<typeof UserInfo>;
```
Comment (PuneetPunamiya) [inline ui/src/store/auth.ts:23]:
```suggestion
export type IUserProfile = Instance<typeof UserInfo>;
```

**Review by PuneetPunamiya — COMMENTED** [review]
Thanks :call_me_hand: 
/approve
/meow
/woof

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Pull request merged**